### PR TITLE
Basecamp Obstacle course

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Basecamp Obstacle Course"

#### Purpose of change
First draft at creating a reasonable way of training the parkour proficiency.  This method goes through the faction camp.  

#### Describe the solution
Adds furniture terrain and fake items to create the basecamp obstacle course which has a fake item that allows training parkour proficiency.   

#### Describe alternatives you've considered
Waiting on someone else.

#### Testing
Tested and it can be built and the practice recipe works.

#### Additional context

parkour

### CODE REVIEW

**Comment by Maleclypse** [comment]
@PatrikLundell This is a first draft but I wanted to request your review since you are probably the person doing the most with faction camps right now.

**Review by PatrikLundell — COMMENTED** [review]
The basecamp.json parkour fake item was a bit hastily constructed, as it would make for bad training if it flattens the trainees on failures... Should be something like "A fake parkour gym used for parkour training in basecamps."

modular_field_common.json:
I don't believe you can have terrain in the furniture section of palettes, and it would conflict with the terrain entries if it was possible. Just omit to provide any furniture section for 'x' and '+' like you do with '.'. Also, I would extend the common palette with new tokens rather than creating a new palette that uses different definitions for ones already in use.

When I try to fit fbmh_2_parkour_south into the camp mentally I get it to overwrite the building at least to the south, and possibly to the east as well. Also, why is it called "_south" when there isn't anything to the north of it, and it's placed in the northern courtyard? I'd just call it "fbmh_2_parkour".

recipe_modular_field_common.json: I'd expand the "description" to "Let's build a parkour obstacle training course in the northern courtyard."
I'd also skip the "blueprint_requires" section to allow the player to build the course whenever they feel like it, rather than gate it by having constructed a lot of the unrelated stuff first, as that's the philosophy I've tried to use for the version 2 things.

Finally, I'd update the layout generated by "fbmh_2" to indicate the future parkour training area's footprint.

This is the first time I've used the "review changes" functionality, so it's somewhat clunky. I've seen others referencing relevant parts, but I don't know how to do that.

I think it's a good use of the courtyard space, though, and I agree with its placement in the northern courtyard so the southern one can remain free for player use (until we come up with enough additions to need to use it).

**Comment by Venera3** [comment]
> This is the first time I've used the "review changes" functionality, so it's somewhat clunky. I've seen others referencing relevant parts, but I don't know how to do that.

In the files changed tab you can hover over the changed line's beginning and a blue + pops - up clicking or dragging that allows you to leave a comment on those lines specifically. In that comment field you will also find a small... I want to say "piece of paper with a plus sign" icon, that allows you to make suggestions the author can apply with a simple confirmation click.

**Comment by Maleclypse** [comment]
> 

OK I've tried to fix all of these except the palette one that I'm going to need to work on and this "Finally, I'd update the layout generated by "fbmh_2" to indicate the future parkour training area's footprint."   I'm thinking that is the design at the top of the file `modular_field_common.json`?

File: data/json/recipes/other/materials.json [hunk 1]
```diff
@@ -45,6 +45,20 @@
...
+    "activity_level": "LIGHT_EXERCISE",
+    "result": "shredded_rubber",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_MATERIALS",
+    "skill_used": "survival",
+    "skills_required": [ "survival", 1 ],
+    "difficulty": 1,
+    "time": "10m",
+    "autolearn": true,
+    "qualities": [ { "id": "CUT", "level": 1  ],
```
Comment (Maleclypse) [inline data/json/recipes/other/materials.json:59]:
```suggestion
    "qualities": [ { "id": "CUT", "level": 1 } ],
```

**Comment by PatrikLundell** [comment]
I think you've made a mistake when converting parkour monkey bars from terrain to furniture. The palette change should still define a terrain component for it as the bars are moved to furniture, but it should be your rubber mulch. Otherwise they'll be placed on whatever terrain was there previously ("t_dirt", as that's what the outline indication makes it).

Also, I realize I've been rather unclear with regards to palette symbols used. The symbols you've used aren't defined here, but they are used in the building construction palettes ('+' = door, 'v' = window, 'w'=wall). While there's no conflict technically, there may still be one in the mind of humans.
Thus, I'd suggest 'b' = bar, 'c' = counter, 'e' = earthbag half, 'E' = earhtbag wall (and remember to adjust both the terrain and the furniture sections: easy to forget one of them).

The image has the course placed one tile lower than the outline, but I believe you corrected the placement in one of the changes.

**Comment by Maleclypse** [comment]
> I think you've made a mistake when converting parkour monkey bars from terrain to furniture. The palette change should still define a terrain component for it as the bars are moved to furniture, but it should be your rubber mulch. Otherwise they'll be placed on whatever terrain was there previously ("t_dirt", as that's what the outline indication makes it).
> 
> Also, I realize I've been rather unclear with regards to palette symbols used. The symbols you've used aren't defined here, but they are used in the building construction palettes ('+' = door, 'v' = window, 'w'=wall). While there's no conflict technically, there may still be one in the mind of humans. Thus, I'd suggest '#' = bar, 'c' = counter, 'e' = earthbag half, 'E' = earhtbag wall (and remember to adjust both the terrain and the furniture sections: easy to forget one of them).
> 
> The image has the course placed one tile lower than the outline, but I believe you corrected the placement in one of the changes.
> 
> Edit: '#' is what the monkey bars look like in the image (and probably in reality, to some extent), so I've changed the recommended symbol above.

Cool thanks.  I think I've made the all the changes you've suggested.

**Comment by PatrikLundell** [comment]
I believe you've made a mistake and pushed the last commit to the wrong branch.

**Pull request merged**